src/main/java/org/folio/circulation/support/SingleRecordFetcher.java

```diff
@@ -58,6 +60,8 @@ public SingleRecordFetcher(
   public CompletableFuture<Result<T>> fetch(String id) {
     log.info("Fetching {} with ID: {}", recordType, id);
 
+    requireNonNull(id, format("Cannot fetch single %s with null ID", recordType));
```

Comment:
```suggestion
    requireNonNull(id, {} -> format("Cannot fetch single %s with null ID", recordType));
```

While a bit uglier, we hope using this message is a rare occurrence, so we should prefer not to construct it unless absolutely necessary. Formatting a string is likely more expensive than the creation of the supplier. No big deal either way, may be "premature optimization" or my part.

Reply:
I prefer the supplier variant of this as well (I think it is more intent revealing, and it is a pity the logger does not have similar). 

In a previous pull request I got the feedback to remove the supplier.

Once again, it comes down to consistency I think, we just need to pick one, document it, and go with it.

Reply:
Hmm... I suppose if it was just a literal String like "cannot be null", then yes, I'd say a supplier is unnecessary. Any sort of string building/formatting, I'd be for putting it in a supplier.

Log4j2 has the ability to provide a supplier, like `log.error("blah {} blah {} blah", () -> expensiveToStringCall(), () -> anotherExpensiveToStringCall());`. It is pretty nice to not have `if (debug) { log.debug("something"); }` all over the place. 😄 It looks like slf4j will handle this in the upcoming 2.0 version using a fluent API. It looks a little wonky though.

Reply:
Does FOLIO have a standard Java logger?

The logging in mod-circulation needs a fair bit of work. Supplier support would be nice. I really don't want to put guard clauses around them all (and that would lead me to writing wrapper functions for the logger)

Reply:
Does FOLIO have a standard Java logger?

The logging in mod-circulation needs a fair bit of work. Supplier support would be nice. I really don't want to put guard clauses around them all (and that would lead me to writing wrapper functions for the logger)

Reply:
I am not aware of a FOLIO standard logging. For example, in edge-sip2, since we were starting from scratch, we used log4j2. Any library we depend on that uses slf4j is redirected to log4j2 using the log4j2 slf4j binding. While I appreciate the value of swapping out the log implementation slf4j provides, how often do we do that?